src/journal-remote/journal-remote-main.c

```diff
@@ -289,10 +311,20 @@ static mhd_result request_handler(
 
         log_trace("Handling a connection %s %s %s", method, url, version);
 
-        if (*connection_cls)
+        if (*connection_cls) {
+                RemoteSource *source = (RemoteSource*)*connection_cls;
```

Comment:
void* does not require explicit casts, it silently auto-casts into any other pointer type